components/infobox/wikis/halo/infobox_league_custom.lua

```diff
@@ -0,0 +1,290 @@
+---
+-- @Liquipedia
+-- wiki=halo
+-- page=Module:Infobox/League/halo
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local League = require('Module:Infobox/League')
+local String = require('Module:String')
+local Template = require('Module:Template')
+local Variables = require('Module:Variables')
+local Tier = require('Module:Tier')
+local PageLink = require('Module:Page')
+local Class = require('Module:Class')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+local _ABBR_USD = '<abbr title="United States Dollar">USD</abbr>'
+local _TODAY = os.date('%Y-%m-%d', os.time())
+local _GAME = mw.loadData('Module:GameVersion')
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_args = league.args
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		return {
+			Cell{name = 'Game version', content = {
+					CustomLeague._getGameVersion()
+				}},
+			}
+	elseif id == 'prizepool' then
+		return {
+			Cell{
+				name = 'Prize pool',
+				content = {CustomLeague:_createPrizepool()},
+			},
+		}
+	elseif id == 'liquipediatier' then
+		return {
+			Cell{
+				name = 'Liquipedia tier',
+				content = {CustomLeague:_createTierDisplay()},
+			},
+		}
+	elseif id == 'customcontent' then
+		--player breakdown
+		if _args.player_number then
+			table.insert(widgets, Title{name = 'Players'})
+			table.insert(widgets, Cell{name = 'Number of players', content = {_args.player_number}})
+		end
+
+		--teams section
+		if _args.team_number or (not String.isEmpty(_args.team1)) then
+			Variables.varDefine('is_team_tournament', 1)
+			table.insert(widgets, Title{name = 'Teams'})
+		end
+		table.insert(widgets, Cell{name = 'Number of teams', content = {_args.team_number}})
+		if not String.isEmpty(_args.team1) then
+			local teams = CustomLeague:_makeBasedListFromArgs('team')
+			table.insert(widgets, Center{content = teams})
+		end
+
+		--maps
+		if not String.isEmpty(_args.map1) then
+			table.insert(widgets, Title{name = 'Maps'})
+			table.insert(widgets, Center{content = CustomLeague:_makeBasedListFromArgs('map')})
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:_createPrizepool()
```

Comment:
Why is this function so complicated? Can we not just use lua formatting and rounding tools?

Reply:
simplified it a bit

that function is a bit more complicated as it first calls a function that cleans the input values (removing currency symbols and abbreviations as well as `,`s in the input) so automatic currency conversion can be used

Reply:
We don't have this on Rocket League, so is it needed there too?

Reply:
it enables automatic conversion

RL atm does not have automatic conversion from local currency to usd via the extension, you just enter the values manually

i think the automatic conversion is a nice feature, but atm it is optional